File: examples/ex6.R
# In order to start server run following line in examples directory
# `R -f ex6.R --gui-none --no-save` - this will block while Rserve is started

# because we run run.Rserve() the current session will become Rserve session
# so we can initialize the server here

# here goes some code you want to load once for example an echo function
echo <- function(data) {
    print(data)
    return(data);
}

# then run Rserve in process
require('Rserve')
run.Rserve()

File: examples/ex6.js
// In order to start server run following line in examples directory
// R -f server.R --gui-none --no-save

"use strict";

var rio = require("../lib/rio");

function printEcho(err, res) {
    if (err) {
        console.log("An error occured : " + err);
    } else {
        console.log("Response is : " + res);
    }
}

rio.e({
    entrypoint: "echo",
    data: ["test", "data"],
    callback: printEcho
});
